cloudcix_primitives/snapshot_lxd.py

```diff
@@ -14,18 +15,309 @@
     'update',
 ]
 
+def build(
+        endpoint_url: str,
+        project: str,
+        container_name: str,
+        snapshot_name: str,
+        stateful: bool = False,
+        verify_lxd_certs: bool = True
+) -> Tuple[bool, str, dict]:
+    """Create a snapshot for an LXD container.
+    
+    :param endpoint_url: The endpoint URL for the LXD Host.
+    :param project: The LXD project name.
+    :param instance_name: The name of the LXD container to snapshot.
+    :param snapshot_name: The name to give the snapshot.
+    :param stateful: Whether to include the runtime state in the snapshot. 
+    NOTE: Stateful Containers require CRIU installation and config migration.stateful = true. 
```

Comment:
This needs to be discussed, but I do not think `stateful` is a required param to build a snapshot of a container. 

1. **Containers require CRIU installation:** Initial research this is required for Live Migration which is not the purpose of a snapshot
2. **config migration.stateful = true:** mitigation.stateful is only a property for virtual machines - https://documentation.ubuntu.com/lxd/en/stable-4.0/instances/